std::span のコンストラクタ呼び出しに関する調整

habu1010 opened the pull request:
Resolves #4911

Hourier reviewed (dismissed):
対応お疲れ様です
ビルドも通っているようですし特に問題なさそうです
Approveします

backwardsEric commented:
These changes applied to 1cba838de1237de4a084805f8d4fd071000bf0e2 compiled with Apple's clang 14 resulted in one error:

```
monster-floor/monster-sweep-grid.cpp:484:68: error: no viable constructor or deduction guide for deduction of template arguments of 'span'
    const auto pos_move = MonsterMoveGridDecider::evalute_deciders(std::span(deciders.begin(), deciders.size()), this->player_ptr->get_position()); // @todo clang-15以降は直接decidersを渡せる
```

I did not have the X11 headers available on the system where that compilation was done so the use of std::span in main-x11.cpp was not tested during the compilation.

habu1010 commented:
> These changes applied to [1cba838](https://github.com/hengband/hengband/commit/1cba838de1237de4a084805f8d4fd071000bf0e2) compiled with Apple's clang 14 resulted in one error:

When I rebased to the latest develop branch, I forgot to fix the std::span constructor call in the updated in the develop branch.
I have fixed it and force pushed it, so could you please try again?

> 
> I did not have the X11 headers available on the system where that compilation was done so the use of std::span in main-x11.cpp was not tested during the compilation.

Unless you are building the X11 version with Xcode 14.2, I don't think you need to worry about it.

backwardsEric commented:
That builds without errors using Apple's clang 14.

habu1010 commented:
@backwardsEric 
Thank you for the confirmation.
We would like to avoid using the constructor of std::span that accepts iterators for the time being so as to avoid any complications from you.

Hourier reviewed (approved):
再Approveします

habu1010 commented:
確認ありがとうございました 🙇🏻
